pkg/openstack/placement.go

```diff
@@ -108,9 +112,6 @@ func ReconcilePlacementAPI(ctx context.Context, instance *corev1beta1.OpenStackC
 		if placementAPI.Spec.Secret == "" {
 			placementAPI.Spec.Secret = instance.Spec.Secret
 		}
-		if placementAPI.Spec.NodeSelector == nil && instance.Spec.NodeSelector != nil {
-			placementAPI.Spec.NodeSelector = instance.Spec.NodeSelector
-		}
```

Comment:
I thought we set it inside the createOrPatch to actually ensure it is updated later not just during initial created.

Reply:
we apply it to the `instance.Spec.Placement.Template` which added via the DeepCopyInto within the `CreateOrPatch` a couple of lines above this. so it should be the same as having this

Reply:
`placementAPI.Spec.NodeSelector == nil` can only be true on the first invocation

Reply:
`DeepCopyInto` will reset it to nil though, so probably did work in this case. IIRC it wasn't correct for some services